### PR TITLE
Ouptut cuda shared memory memory cleanup

### PR DESCRIPTION
**Pull request by CoderHam**
Solves #1001

### CODE REVIEW

**Comment by GuanLuo** [comment]
How does this PR solve #1001

**Comment by CoderHam** [comment]
The temp buffer (tmp_addr) used to copy the data from cuda shared memory to host is not needed and is never cleared. The change I made above removes the need for this temporary buffer.

File: src/clients/python/__init__.py [hunk 1]
```diff
@@ -1416,53 +1417,62 @@ def _get_results(self, outputs, batch_size, request_id=None):
...
-                        cval = shm_addr
-                        for b in range(batch_size):
-                            cval_len = start_pos + element_byte_size
-                            if cval_len == 0:
-                                val = np.empty(shape, dtype=result_dtype)
-                                results[output_name].append(val)
-                            else:
-                                val_buf = cast(cval, POINTER(c_byte * cval_len))[0]
-                                val = np.frombuffer(val_buf, dtype=result_dtype, offset=start_pos)
+                    try:
```
Comment (GuanLuo) [inline src/clients/python/__init__.py]: Why `try` block here? I don't see there are early returns in the `try` block

Comment (GuanLuo) [reply src/clients/python/__init__.py]: Or you should move `try` up to cover `_crequest_get_shared_memory_handle_info`. Note that exception may be raised from `_crequest_get_shared_memory_handle_info` after the internal buffer is allocated

Comment (CoderHam) [reply src/clients/python/__init__.py]: We allocate a new memory buffer (in the case of GPU shared memory) in _crequest_get_shared_memory_handle_info. If for some reason there is an exception while processing that into a numpy array, we still want this buffer to be destroyed.

Comment (GuanLuo) [reply src/clients/python/__init__.py]: > Or you should move `try` up to cover `_crequest_get_shared_memory_handle_info`. Note that exception may be raised from `_crequest_get_shared_memory_handle_info` after the internal buffer is allocated

Did you do this?

Comment (CoderHam) [reply src/clients/python/__init__.py]: Done

**Pull request merged**